x/metadata/client/cli/tx.go

```diff
@@ -242,3 +246,148 @@ func ModifyOsLocatorCmd() *cobra.Command {
+			}
+
+			partiesInvolved := parsePartyTypes(args[2])
+			description := parseDescription(args[7:])
+			contractSpecification := types.ContractSpecification{SpecificationId: specificationID,
+				Description:     description,
+				OwnerAddresses:  strings.Split(args[1], ","),
+				PartiesInvolved: partiesInvolved,
+				ClassName:       args[5],
+			}
+			switch s := strings.ToUpper(args[3]); s {
+			case "RESOURCEID":
+				var recordID types.MetadataAddress
+				recordID, err = types.MetadataAddressFromBech32(args[4])
+				if err != nil {
+					return err
+				}
+				contractSpecification.Source = &types.ContractSpecification_ResourceId{
+					ResourceId: recordID,
+				}
+			case "HASH":
+				contractSpecification.Source = &types.ContractSpecification_Hash{
+					Hash: args[4],
+				}
+			default:
+				return fmt.Errorf("incorrect source type for contract specification: %s", s)
+			}
+
+			signers := strings.Split(args[6], ",")
+
+			msg := types.NewMsgAddContractSpecificationRequest(contractSpecification, signers)
+			err = msg.ValidateBasic()
+			if err != nil {
+				return err
+			}
+
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), msg)
+		},
+	}
+
+	flags.AddTxFlagsToCmd(cmd)
+
+	return cmd
+}
+
+func parsePartyTypes(delimitedPartyTypes string) []types.PartyType {
+	parties := strings.Split(delimitedPartyTypes, ",")
+	partyTypes := make([]types.PartyType, len(parties))
+	for i, party := range parties {
+		partyValue := types.PartyType_value[fmt.Sprintf("PARTY_TYPE_%s", strings.ToUpper(party))]
+		partyTypes[i] = types.PartyType(partyValue)
+	}
+	return partyTypes
+}
+
+func parseDescription(cliArgs []string) *types.Description {
+	if len(cliArgs) == 0 {
+		return nil
+	}
+
+	description := types.Description{}
+	if len(cliArgs) >= 1 {
+		description.Name = cliArgs[0]
+	}
+	if len(cliArgs) >= 2 {
+		description.Description = cliArgs[1]
+	}
+	if len(cliArgs) >= 3 {
+		description.WebsiteUrl = cliArgs[2]
+	}
+	if len(cliArgs) >= 4 {
+		description.IconUrl = cliArgs[3]
+	}
+	return &description
+}
+
+// RemoveContractSpecificationCmd creates a command to remove a contract specification
+func RemoveContractSpecificationCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "remove-contract-specification [specification-id] [signers]",
+		Short: "Removes a contract specification on the provenance blockchain",
+		Args:  cobra.ExactArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			var specificationID types.MetadataAddress
+			specificationID, err = types.MetadataAddressFromBech32(args[0])
+			if err != nil {
+				return err
+			}
+
+			if !specificationID.IsContractSpecificationAddress() {
+				return fmt.Errorf("invalid contract specification id: %s", args[0])
+			}
+
+			msg := types.NewMsgDeleteContractSpecificationRequest(specificationID, strings.Split(args[1], ","))
```

Comment:
Does this signers approach work?  Have you tested this command against a localnet-start instance for example?

Reply:
Updated it to user a signers flag if present.